spec/controllers/users_controller_spec.rb

```diff
@@ -25,56 +25,103 @@
+        alice = Fabricate(:user)
+        invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
+        post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
+        expect(Invitation.first.token).to be_nil
+      end
+
+      it "redirects to home_path if @user is saved" do 
+        post :create, user: Fabricate.attributes_for(:user)
+        expect(response).to redirect_to home_path
+      end
 
-    it "makes the inviter follow the user" do
-      alice = Fabricate(:user)
-      invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
-      post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
-      joe = User.where(email: 'joe@example.com').first
-      expect(alice.follows?(joe)).to be_truthy
     end
     
-    it "expires the invitation upon acceptance" do
-      alice = Fabricate(:user)
-      invitation = Fabricate(:invitation, inviter: alice, recipient_email: 'joe@example.com')
-      post :create, user: { email: 'joe@example.com', password: "password", username: 'Joe Doe'}, invitation_token: invitation.token
-      expect(Invitation.first.token).to be_nil
+    context "valid personal info and declined card" do
+      it "does not create a new user record" do
+        charge = double(:charge, successful?: false, error_message: "Your card was declined.")
+        StripeWrapper::Charge.should_receive(:create).and_return(charge)
+        post :create, user: Fabricate.attributes_for(:user), stripeToken: '1231241'
+        expect(User.count).to eq(0)
+      end
+      it "renders the new template" do
+        charge = double(:charge, successful?: false, error_message: "Your card was declined.")
+        StripeWrapper::Charge.should_receive(:create).and_return(charge)
+        post :create, user: Fabricate.attributes_for(:user), stripeToken: '1231241'
+        expect(response).to render_template :new
+      end
+      it "sets the flash error message" do
+        charge = double(:charge, successful?: false, error_message: "Your card was declined.")
+        StripeWrapper::Charge.should_receive(:create).and_return(charge)
+        post :create, user: Fabricate.attributes_for(:user), stripeToken: '1231241'
+        expect(flash[:error]).to be_present
+      end
+      it "does not send out email with invalid inputs" do
+        post :create, user: { email: "joe@example" }
+        expect(ActionMailer::Base.deliveries).to be_empty
+      end
     end
 
-    it "redirects to home_path if @user is saved" do 
-      post :create, user: Fabricate.attributes_for(:user)
-      expect(response).to redirect_to home_path
-    end
+    context "with invalid personal info" do
+
+      it "render :new if @user is not saved" do
+        post :create, user: {email: "gggggg@ggg.com", password: "ekekekekek"}
```

Comment:
It isn't clear here why you are using the email or the password you are. This is the type of gibberish I throw at my server when I'm randomly testing stuff. This is in the code base and therefore the content should either mean something or by not meaning something indicate it is mundane. Is `ggggg@ggg.com` not a good email? Or if it is I would use a clearly valid example. Is the password bad? If so why? Remember you are testing the failing path not the whole set of your user validations. TL;DR Don't make your brain work so hard trying to remember why you used these values.